### PR TITLE
Fixes issue #5178 EditorConfig settings not applied to new file added

### PR DESCRIPTION
**Pull request by mkrueger**
to project

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.6

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpFormatter.cs [hunk 1]
```diff
@@ -114,13 +115,49 @@ public static string FormatText (Microsoft.CodeAnalysis.Options.OptionSet option
...
 		{
 			var chain = DesktopService.GetMimeTypeInheritanceChain (mimeType);
 			var policy = policyParent.Get<CSharpFormattingPolicy> (chain);
 			var textPolicy = policyParent.Get<TextStylePolicy> (chain);
+			var optionSet = policy.CreateOptions (textPolicy);
+
+			if (input is IReadonlyTextDocument doc) {
+				try {
+					var conventions = EditorConfigService.GetEditorConfigContext (doc.FileName).WaitAndGetResult ();
+					Console.WriteLine (doc.FileName +":ctx=" + conventions);
```
Comment (slluis) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpFormatter.cs]: Debug code.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs [hunk 2]
```diff
@@ -43,8 +43,6 @@ static class EditorConfigService
 
 		public static Task<ICodingConventionContext> GetEditorConfigContext (string fileName, CancellationToken token = default (CancellationToken))
 		{
-			if (!File.Exists (fileName))
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs]: What about files which are not written to disk yet (temporary files) or empty file names?

Docs on TextEditor.FileName which is used as an argument for some of the calls: http://source.monodevelop.com/#MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs,388

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs]: Needs to work for files that are not on disk - that's the purpose of it.

**Pull request merged**